test/Microsoft.AspNetCore.Sockets.Tests/TestTaskExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Sockets.Tests
+{
+    public static class TestTaskExtensions
```

Comment:
I started working on modifying xunit to support timeouts, but it was too difficult.